### ISSUE TITLE
Implement create a new issue

### ISSUE THREAD

**Issue by jordilin** [jordilin/gitar#170]
```
gr is create --title <title> --description <description>
```

Interface could be close to when creating a new pull request allowing user to specify a description from a file for example.

**Comment by BipulLamsal** [comment]
please guide me on this, i want to work in this issue

**Comment by jordilin** [comment]
Yes, there are several parts. For now I would suggest implementing create a Github issue. It's REST API is https://docs.github.com/en/rest/issues/issues?apiVersion=2022-11-28#create-an-issue 

1. Create cli interface in the cli module (subfolder) - Follow same approach as the other commands, one file per command. That would be a `cli/issue.rs`. See the other commands like `cmd/merge_request` subcommand `comment merge request` for guidance. You'll need to implement the From trait to build an `IssueCliArgs`. The `IssueCliArgs` struct can be placed in `cmd/issue.rs` 
2. Add a new trait in `api_traits.rs` called `Issue`. For now this would be just one operation `create`. `fn create(&self, args: IssueBodyArgs) -> Result<Issue>` The `IssueBodyArgs` should carry the necessary information such as the `title` and the `description` of the issue. Implementors such as `Github` will retrieve the information from the Body and construct the URL as specified in the docs.
3. We will need a `cmd/issue.rs` with an `execute` function (see other commands) where we take the remote implementing the `Issue` trait, construct an `IssueBodyArgs` based on an `IssueCliArgs`. The remote implementing the trait can be created using the `get` macro in remote module. Ex https://github.com/jordilin/gitar/blob/main/src/remote.rs#L579
4. Implement Issue for Github - Create module `github/issue.rs`. Similar approach as https://github.com/jordilin/gitar/blob/main/src/github/merge_request.rs#L257 but we will get an `Issue` back instead. The Issue struct can be defined in `cmd/issue.rs` and we will need to get its URL and ID - Both ID and URL come in the json response. You can gather the example response in the docs and place it under `contracts/github` folder `issue.json`  - You can use the `get_contract` test helper to read the contents and use it to verify the json response gets properly mapped. There are plenty of examples around in the code. 
5. The `remote/query.rs` contains macros where we can define functions to be used from within the trait implementation. This is a single call, so we need to use the `send` macro. Example https://github.com/jordilin/gitar/blob/main/src/remote/query.rs#L326 where it creates a function and maps Github specific fields to the struct we need in the response. In that case we need an `Issue`. For the ApiOperation argument you can use `ApiOperation::Project` for now.
6. The user should see something like `Issue ID <id> with URL <url> has been created`.
7. Tests are required for the cli interface, cmd and github - Check similar examples in the code.

I'll probably add documentation on how the application is structured, but for now this should get you going. Thanks.
